R/NetStats.R

```diff
@@ -168,16 +168,40 @@ build_netstats <- function(epistats, netparams,
   out$attr$role.class <- attr_role.class
 
   # diag status
-  if (race == TRUE) {
-    xs <- data.frame(age = attr_age, race.cat3 = attr_race, geogYN = 1)
-    preds <- predict(epistats$hiv.mod, newdata = xs, type = "response")
-    attr_diag.status <- rbinom(num, 1, preds)
-    out$attr$diag.status <- attr_diag.status
-  }  else {
-    xs <- data.frame(age = attr_age, geogYN = 1)
-    preds <- predict(epistats$hiv.mod, newdata = xs, type = "response")
-    attr_diag.status <- rbinom(num, 1, preds)
-    out$attr$diag.status <- attr_diag.status
+  if (is.null(epistats$init.hiv.prev)) {
+    if (race == TRUE) {
+      xs <- data.frame(age = attr_age, race.cat3 = attr_race, geogYN = 1)
+      preds <- predict(epistats$hiv.mod, newdata = xs, type = "response")
+      attr_diag.status <- rbinom(num, 1, preds)
+      out$attr$diag.status <- attr_diag.status
+    }  else {
+      xs <- data.frame(age = attr_age, geogYN = 1)
+      preds <- predict(epistats$hiv.mod, newdata = xs, type = "response")
+      attr_diag.status <- rbinom(num, 1, preds)
+      out$attr$diag.status <- attr_diag.status
+    }
+  } else {
+    if (race == TRUE) {
+      init.hiv.prev <- epistats$init.hiv.prev
+      samp.B <- which(attr_race == 1)
+      samp.H <- which(attr_race == 2)
+      samp.W <- which(attr_race == 3)
+
+      attr_diag.status <- rep(0, network.size)
+
+      attr_diag.status[samp.B] <- rbinom(length(samp.B), 1, init.hiv.prev[1])
+      attr_diag.status[samp.H] <- rbinom(length(samp.H), 1, init.hiv.prev[2])
+      attr_diag.status[samp.W] <- rbinom(length(samp.W), 1, init.hiv.prev[3])
+
+      out$attr$diag.status <- attr_diag.status
+
+    } else {
+    init.hiv.prev <- epistats$init.hiv.prev
+    samp.size <- ceiling(network.size*init.hiv.prev)
```

Comment:
As previously discussed, you need to use standardized code styling (e.g., this block should be indented)